### ISSUE TITLE
Logic bug in bmi2.c

### ISSUE THREAD

**Issue by thealastair** [BoschSensortec/BMI270-Sensor-API#2]
Line 4691 - 
```
  if ((sens_cfg[loop].type != BMI2_ACCEL) || (sens_cfg[loop].type != BMI2_GYRO) ||
                (sens_cfg[loop].type != BMI2_TEMP) || (sens_cfg[loop].type != BMI2_AUX))
```
GCC correctly detects this as a bug
error: logical 'or' of collectively exhaustive tests is always true [-Werror=logical-op]

I assume the ||'s are supposed to be && ?

**Comment by thealastair** [comment]
See https://github.com/BoschSensortec/BMI270-Sensor-API/pull/3

**Comment by BoschSensortec** [comment]
Fixed in v2.53.2
